doc/nrf/templates/library_template_README.rst

```diff
@@ -0,0 +1,143 @@
+.. _library_template:
+
+Template: Library
+#################
+
+.. contents::
+   :local:
+   :depth: 2
+
+.. note::
+   Use this template to create pages that describe the libraries included in |NCS|, following these instructions:
+
+   * Use this first section to give a short description (1 or 2 sentences) of the library, indicating what it does and why it should be used.
+
+   * Give the library a concise name that also corresponds to the folder name.
+
+     If the library targets a specific protocol or device, add it in the title before the library name (for example, "NFC:" or "nRF9160:").
+     Do not include the word "library" in the title.
+     Use the provided stock phrases and includes when possible.
+
+   * Sections with ``*`` are optional and can be left out.
+
+     All the other sections are required for all libraries.
+     Do not add new sections or remove a mandatory section without consulting a tech writer.
+     If any new section is needed, try first to fit it as a subsection of one of the sections listed below.
+     When you keep an optional section, remove the asterisk from the title, and format the length of the related RST heading accordingly.
+
+   * Include a tech writer in the PR review.
+
+Overview
+********
+
+.. note::
+   Use this section to give a general overview of the library.
+
+Implementation
+==============
+
+.. note::
+   Use this section to describe the library architecture and implementation.
+
+Supported features
+==================
+
+.. note::
+   Use this section to describe the features supported by the library.
+
+Supported backends*
+===================
+
+.. note::
+   Use this section to describe the backends supported by the library, if applicable.
+   This is optional.
+
+Requirements*
+*************
+
+.. note::
+   Use this section to list the requirements needed to use the library.
+   This is not required if there are no specific requirements to use the library. It is a mandatory section if there are specific requirements that have to be met to use the library.
+
+Library files
+*************
+
+.. note::
+   Use this section to indicate the location of the library files.
+
+This library can be found under :file:`path_to_library_files` in the |NCS| folder structure.
+
+Configuration*
+**************
+
+.. note::
+   Use this section to list available library configuration options, if applicable.
+   This is optional.
+
+Shell commands list*
+====================
+
+.. note::
+   Use this section to list available library shell commands, if applicable.
+   This is optional.
+
+Usage
+*****
+
+.. note::
+   Use this section to explain how to use the library.
+
+Samples using the library*
```

Comment:
Could this be automatized somewhat? For example, we could have a script that would run a check of all dependencies on sample pages and check if the given sample page is referenced on the related library page...

Reply:
I honestly don't think this should be within the scope of this PR. We would have to ask Gerard about that.